concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_discarding_conversion.rs

```diff
@@ -0,0 +1,238 @@
+    ///     DecompositionBaseLog, DecompositionLevelCount, GlweDimension, LweDimension, PolynomialSize,
+    /// };
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(256);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input = 3_u64 << 50;
+    /// let noise = Variance(2_f64.powf(-25.));
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let key_1: GlweSecretKey64 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    /// let plaintext = engine.create_plaintext(&input)?;
+    ///
+    /// let mut ciphertext =
+    ///     engine.encrypt_ggsw_ciphertext(&key_1, &plaintext, noise, level, base_log)?;
+    ///
+    /// let mut fourier_ciphertext: GgswCiphertextComplex64 =
+    ///     engine.convert_ggsw_ciphertext(&ciphertext)?;
+    ///
+    /// // We're going to re-encrypt and re-convert the input with another secret key
+    /// // For this, it is required that the second secret key uses the same GLWE dimension
+    /// // and polynomial size as the first one.
+    /// let key_2: GlweSecretKey64 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    ///
+    /// engine.discard_encrypt_ggsw_ciphertext(&key_2, &mut ciphertext, &plaintext, noise)?;
+    /// engine.discard_convert_ggsw_ciphertext(&mut fourier_ciphertext, &ciphertext)?;
+    ///
+    /// #
+    /// assert_eq!(fourier_ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(fourier_ciphertext.polynomial_size(), polynomial_size);
+    /// assert_eq!(fourier_ciphertext.decomposition_base_log(), base_log);
+    /// assert_eq!(fourier_ciphertext.decomposition_level_count(), level);
+    ///
+    /// engine.destroy(key_1)?;
+    /// engine.destroy(key_2)?;
+    /// engine.destroy(plaintext)?;
+    /// engine.destroy(ciphertext)?;
+    /// engine.destroy(fourier_ciphertext)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn discard_convert_ggsw_ciphertext(
+        &mut self,
+        output: &mut GgswCiphertextComplex64,
+        input: &GgswCiphertext64,
+    ) -> Result<(), GgswCiphertextDiscardingConversionError<Self::EngineError>> {
+        GgswCiphertextDiscardingConversionError::perform_generic_checks(output, input)?;
+        unsafe { self.discard_convert_ggsw_ciphertext_unchecked(output, input) };
+        Ok(())
+    }
+
+    unsafe fn discard_convert_ggsw_ciphertext_unchecked(
+        &mut self,
+        output: &mut GgswCiphertextComplex64,
+        input: &GgswCiphertext64,
+    ) {
+        let input_bsk = StandardBootstrapKey::from_container(
+            input.0.as_tensor().as_container().clone(),
```

Comment:
Do we actually need to clone the input here?

Reply:
To perform the conversion to the Fourier domain, I wanted to use the fill_with_forward_fourier method only available in a BSK, hence the conversion to and from BSK. The clone is needed because GgswCiphertext are not Copy. 
There is probably a better, cleaner way to proceed. Maybe by adding a fill_with_forward_fourier method inside private/ggsw?